### ISSUE TITLE
出现Connection aborted错误

### ISSUE THREAD

**Issue by carrytameng** [AirtestProject/Poco#16]
环境：
mac
android7.1.1

实现功能：一直下拉加载更多
在下拉一定时间后会导致如下错误：

```
[2018-04-28 06:09:02][ERROR] ('Connection aborted.', BadStatusLine("''",))
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@5a4eb9c>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@68ccf0f>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@7ef1c6e>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@c64d7e9>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@7d4b870>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@1a33ab3>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of [Lcom.netease.open.libpoco.sdk.AbstractNode;@e1b2022>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of [Lcom.netease.open.libpoco.sdk.AbstractNode;@af4917a>'> ignored
[2018-04-28 06:09:02][ERROR] ('Connection aborted.', error(54, 'Connection reset by peer'))
```
求大神们如何解决此问题

**Comment by adolli** [comment]
这个是由于用于检测Hierarchy的PocoService后台进程意外被系统中断导致的，另一种可能是你在脚本中引用了过多远程临时变量。可以贴一下你的脚本怎么写的吗？

**Comment by carrytameng** [comment]
from emulate_util import *
from emulate_init import *

# get call detail
def get_call_detail(poco):
    log.warning('get call detail start')
    try:
        if check_and_click(poco(desc="切换日期")):
            if check_and_click(poco(desc="按月查询")):
                months = wait_and_get(poco("pageNew").child("android.view.View")[1].offspring("android.widget.ListView")[1].child("android.view.View"))
                month_len = len(months)
                check_and_click(poco(desc="切换日期"))
                for i in range(month_len):
                    try:
                        if not loop_click(poco, i):
                            loop_click(poco, i)
                    except Exception as e:
                        log.exception(e)
                log.warning('get call detail end')
                return True
    except Exception as e:
        log.exception(e)
    log.warning('get call detail error')
    return False

# loop click the date calendar
def loop_click(poco, i):
    log.warning('loop click start: ' + str(i))
    try:
        if wait_and_click(poco(desc="切换日期")):
            if wait_and_get(poco(desc="按月查询")):
                months = wait_and_get(poco("pageNew").child("android.view.View")[1].offspring("android.widget.ListView")[1].child("android.view.View"))
                check_and_click(months[i])
                while True:
                    month_flag = wait_and_get(poco(descMatches='.*月的通话详单如下.*'))
                    if month_flag:
                        log.info(month_flag.attr('desc'))
                        break
                    sleep(0.1)
                swipe_to_bottom(poco)
                log.warning('loop click end: ' + str(i))
                return True
    except Exception as e:
        log.exception(e)
    log.warning('loop click error: ' + str(i))
    return False

# swipe to bottom
def swipe_to_bottom(poco):
    while True:
        if check_and_click(poco(descMatches='.*加载更多.*')):
            pass
        elif check_and_click(poco("com.sinovatech.unicom.ui:id/custom_dialog_cancel_button")):
            pass
        elif not poco(desc="已加载全部").exists():
            poco.swipe([0.9, 0.9], [0.9, 0.1], duration=0.05)
            sleep(1)
        elif poco(desc="已加载全部").exists():
            # get_all_element_attr(poco("pageNew").child("android.view.View")[1].offspring("android.view.View"))
            while not poco(descMatches='.*月的通话详单如下.*').exists():
                poco.swipe([0.9, 0.25], [0.9, 0.9], duration=0.05)
                sleep(0.1)
            break
        else:
            sleep(0.1)

if __name__ == '__main__':
    from emulate_init import *

    device, poco = init_device()
    get_call_detail(poco)



![image](https://user-images.githubusercontent.com/5178758/39567095-6a108b12-4ef0-11e8-9aa6-e55a70b3d9be.png)

![image](https://user-images.githubusercontent.com/5178758/39567110-787520be-4ef0-11e8-8b22-094ed109a992.png)

**Comment by adolli** [comment]
大概运行了多长时间才报错的？另外可以尝试更新下`pip install -U pocoui`

**Comment by carrytameng** [comment]
pocoui已更新，时间应该5分钟左右。最近测试另一个不断下拉列表也出现这种情况：

Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@5c62395>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@2fa774c>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@dfff49e>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@22e32d9>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@e2fe920>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@87f0023>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@efdd552>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@db115dd>'> ignored
Exception requests.exceptions.ConnectionError: ConnectionError(ProtocolError('Connection aborted.', error(54, 'Connection reset by peer')),) in <bound method RpcObjectProxy.__del__ of u'<Rpc remote object proxy of com.netease.open.libpoco.Node@e72bcaa>'> ignored

**Comment by adolli** [comment]
fixed at `pocoui==1.0.67`. try to upgrade by `pip install -U pocoui`.
